packages/runtime-core/src/component.ts

```diff
@@ -814,6 +814,7 @@ export function finishComponentSetup(
           // pass runtime compat config into the compiler
           finalCompilerOptions.compatConfig = Object.create(globalCompatConfig)
           if (Component.compatConfig) {
+            // @ts-expect-error types are not compatible
             extend(finalCompilerOptions.compatConfig, Component.compatConfig)
```

Comment:
I've checked the types and they are different, we could possible update the `finalCompilerOptions.compatConfig` to also contain the `Component.compatConfig`, but I wasn't sure about it